### PR TITLE
feat: implement redis database number

### PR DESCRIPTION
**Pull request by legioner0**
# Pull Request Template

## Description

This PR adds configurable Redis database number.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Build and start app in docker with custom `REDIS_DB_NUM` value - works correctly.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -28,6 +28,8 @@ REDIS_URL=redis://redis:6379
 # which will be the password for the redis service running inside the docker-compose
 # to make it secure
 REDIS_PASSWORD=
+# Database number, default 1
```
Comment (sojan-official) [inline .env.example:31]: Think the default db which rails use is 0

File: config/cable.yml [hunk 2]
```diff
@@ -2,7 +2,8 @@ default: &default
   adapter: redis
   url: <%= ENV.fetch('REDIS_URL', 'redis://127.0.0.1:6379') %>
   password: <%= ENV.fetch('REDIS_PASSWORD', nil).presence %>
-  ssl_params: 
+  db: <%= ENV.fetch('REDIS_DB_NUM', 1) %>
```
Comment (sojan-official) [inline config/cable.yml:5]: the default value should be 0

File: spec/lib/redis/config_spec.rb [hunk 3]
```diff
@@ -4,19 +4,22 @@
   context 'when single redis instance is used' do
     let(:redis_url) { 'redis://my-redis-instance:6379' }
     let(:redis_pasword) { 'some-strong-password' }
+    let(:redis_db_num) { '2' }
```
Comment (sojan-official) [inline spec/lib/redis/config_spec.rb:7]: let's take the case where Redis db number is specified as a separate case in the specs and ensure the default one remains 0 when environment variable is not present

**Review by sojan-official — COMMENTED** [review]
- Few review comments 

ref: 

https://gitlab.com/gitlab-org/omnibus-gitlab/-/issues/454
https://chat.openai.com/share/4ebcdaf3-79aa-495d-b25b-c53ac7197eed

File: spec/lib/redis/config_spec.rb (hunk 3 above)
Comment (legioner0) [reply spec/lib/redis/config_spec.rb]: done

**Comment by sojan-official** [comment]
Hi @legioner0,

Thank you for your PR and the effort you've put into this! We discussed this update with the development team and realized that the current setup already supports configuring a database number through the URL option, like so:

```
REDIS_URL=redis://localhost:6379/2
```

Given that this functionality can already be achieved through existing configurations and considering the niche use case, we believe adding a new configuration option might not be necessary at this time. Therefore, we'll be closing this PR. Thanks again for your contribution and understanding!

**Review by sojan-official — APPROVED** [review]

**Pull request closed without merging**